### ISSUE TITLE
Mouseup does not fire outside of webview after mousedown in webview

### ISSUE THREAD

**Issue by krisdages** [electron/electron#17635]
### Preflight Checklist

* [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 4.1.2
  * Latest 5.0 beta

* **Operating System:**
  * All
* **Last Known Working Electron version:**:
  * 3.x

### Expected Behavior
When dragging inside of a webview and releasing the mouse button outside, the mouseup event should fire on the `window` object inside of the webview.

### Actual Behavior
The webview's window never receives the mouseup event. This breaks any library that relies on receiving the mouseup (such as D3) if the user accidentally overshoots the edge of the webview before releasing the mouse.

This behavior does not occur with an OOPIF as far as I can tell using Chrome,
and it did not happen with Electron 3 and before.

### To Reproduce
https://github.com/krisdages/electron-issue-webview-mouseevents

```
# From terminal:
git clone https://github.com/krisdages/electron-issue-webview-mouseevents
cd electron-issue-webview-mouseevents
npm install 
# Run the app
npm start

### Additional Information
The demo uses Electron 4, but the issue persists with the 5.0 beta.

**Comment by krisdages** [comment]
This issue is pretty critical for my project; is anyone aware of the cause? 

If the source and fix are JavaScript-only, I'd be glad to take a crack at it if I could get a pointer in the right direction.

Thanks!

**Comment by krisdages** [comment]
I noticed in the latest release notes that Electron 4 is nearing end of support.  My project, which uses webviews extensively in a tiled layout like Codepen/Electron Fiddle, is still using Electron 3 because of this issue.  Was hoping to find out if there might be any movement on this one so we can get on a recent version. 

tl;dr:
I found a possible workaround and created a new repo with a rough demonstration of it. 
https://github.com/krisdages/electron-17635-webview-mouse-events

--

@deepak1556 I saw some comments in this issue about a potential workaround involving synthetic mouse events for another OOPIF-related problem: https://github.com/microsoft/vscode/issues/75090

I tried the same approach, but mapbox didn't recognize the mouse event fired by `window.dispatchEvent(new MouseEvent(...))`; so I decided to try `webContents.sendInputEvent` instead, and that worked like a charm.

If there is anything I can do to help get a more robust fix for this behavior into the next version, please let me know. 

If there are any problems with an approach like this or if it is unlikely that the Electron team will have the bandwidth to decide on this one in the near future, it would be great to get a heads up so I can plan to implement a workaround locally in my project. 

Thanks again,

Kris 

p.s - if I'm going about this atypically or not following etiquette pls don't hesitate to say so; still a greenhorn when it comes to contributing, sorry.

**Comment by lslzl3000** [comment]
I have the same issue in my project with electron 6.x
No mouseup/mousemove event after sendinputevent in a webview which hold a webgl canvas webpage.
Any work around to solve this problem?

**Comment by yuer0608** [comment]
I have the similar issue in my project with electron 5.0.12.
when mouse move inside webview ，The webview's window never receives the  any mouse event（mousemove、mousedown、mouseup）
I know set webview  pointer-events CSS property to none,then webview's window will receives the  mouse event, but the embed a web page of webview have not receive the mouse event include click.
I want the webview's window  and the embed a web page of webview receive the mouse event, how should I do?

**Comment by krisdages** [comment]
@lslzl3000 @yuer0608 I found this workaround that seems to work for most of my use cases while this issue is still outstanding.

https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture

```ts
const PASSIVE_CAPTURE = Object.freeze({ passive: true, capture: true });
/**
 * Set/release PointerCapture on mousedown/mouseup
 * to prevent mouse drags from breaking when the cursor exits the webview.
 * This is sort of a "blunt object" solution, make sure it doesn't interfere with any other expected behavior.
 */
export function subscribeFixWebviewPointerDrag(): TeardownLogic {
	const body = document.body;
	const win = window;

	const captureHandler = (event: PointerEvent) => {
		body.setPointerCapture(event.pointerId);
		//logger.localWarn("setting pointer capture", event.type);
	};
	const releaseHandler = (event: PointerEvent) => {
		body.releasePointerCapture(event.pointerId);
		//logger.localWarn("releasing pointer capture", event.type);
	};

	win.addEventListener("pointerdown", captureHandler, PASSIVE_CAPTURE);
	win.addEventListener("pointerup", releaseHandler, PASSIVE_CAPTURE);

	//unsubscribe
	return () => {
		win.removeEventListener("pointerdown", captureHandler, PASSIVE_CAPTURE);
		win.removeEventListener("pointerup", releaseHandler, PASSIVE_CAPTURE);
	};
}
```

**Comment by huyinghuan** [comment]
I have the similar issue in my project with electron 10.0.1

![image](https://user-images.githubusercontent.com/3005433/93450644-c8ac2180-f908-11ea-94e1-90a712ab617c.png)


```
module.exports = function (first, element, second, direction) {

    var md; // remember mouse down info    
    element.onmousedown = onMouseDown;

    function onMouseDown(e) {
        console.log("mouse down: " + e.clientX);
        md = {
            e,
            offsetLeft: element.offsetLeft,
            offsetTop: element.offsetTop,
            firstWidth: first.offsetWidth,
            secondWidth: second.offsetWidth
        };
        document.onmousemove = onMouseMove;
        document.onmouseup = () => {
            console.log("mouse up");
            document.onmousemove = document.onmouseup = null;
        }
    }

    function onMouseMove(e) {
        console.log("mouse move: " + e.clientX);
        var delta = {
            x: e.clientX - md.e.clientX,
            y: e.clientY - md.e.clientY
        };

        if (direction === "H") // Horizontal
        {
            // prevent negative-sized elements
            delta.x = Math.min(Math.max(delta.x, -md.firstWidth),
                md.secondWidth);

            element.style.left = md.offsetLeft + delta.x + "px";
            first.style.width = (md.firstWidth + delta.x) + "px";
            second.style.width = (md.secondWidth - delta.x) + "px";
        }
    }
}
```

**Comment by HyperClockUp** [comment]
same issue in 9.0.4

**Comment by Rabbitzzc** [comment]
> I have the similar issue in my project with electron 10.0.1
> 
> ![image](https://user-images.githubusercontent.com/3005433/93450644-c8ac2180-f908-11ea-94e1-90a712ab617c.png)
> 
> ```
> module.exports = function (first, element, second, direction) {
> 
>     var md; // remember mouse down info    
>     element.onmousedown = onMouseDown;
> 
>     function onMouseDown(e) {
>         console.log("mouse down: " + e.clientX);
>         md = {
>             e,
>             offsetLeft: element.offsetLeft,
>             offsetTop: element.offsetTop,
>             firstWidth: first.offsetWidth,
>             secondWidth: second.offsetWidth
>         };
>         document.onmousemove = onMouseMove;
>         document.onmouseup = () => {
>             console.log("mouse up");
>             document.onmousemove = document.onmouseup = null;
>         }
>     }
> 
>     function onMouseMove(e) {
>         console.log("mouse move: " + e.clientX);
>         var delta = {
>             x: e.clientX - md.e.clientX,
>             y: e.clientY - md.e.clientY
>         };
> 
>         if (direction === "H") // Horizontal
>         {
>             // prevent negative-sized elements
>             delta.x = Math.min(Math.max(delta.x, -md.firstWidth),
>                 md.secondWidth);
> 
>             element.style.left = md.offsetLeft + delta.x + "px";
>             first.style.width = (md.firstWidth + delta.x) + "px";
>             second.style.width = (md.secondWidth - delta.x) + "px";
>         }
>     }
> }
> ```

我也是需要这个能力呀
